CMakeLists.txt

```diff
@@ -294,6 +295,42 @@ else()
     message( STATUS "${red}No HIP/ROCm support: gpu_backend = ${gpu_backend}${plain}" )
 endif()
 
+#-------------------------------------------------------------------------------
+# OneMKL/SYCL support.
+message( "" )
+set( blaspp_use_onemkl false )  # output in blasppConfig.cmake.in
+if (NOT gpu_backend MATCHES "^(cuda|hip)$"
+    AND gpu_backend MATCHES "^(onemkl|auto)$" )
```

Comment:
Does this `NOT gpu_backend MATCHES "^(cuda|hip)$"` condition add anything? It doesn't seem possible for a string to match both "^(onemkl|auto)$" and "^(cuda|hip)$".

I think instead it needs to check `NOT blaspp_use_cuda AND NOT blaspp_use_hip`, to make sure that `auto` didn't already find CUDA or ROCm.

Reply:
If `auto` finds cuda or hip, then gpu_backend gets set.  
So I should just need to make sure that gpu_backend is `onemkl|auto`
```
# if cuda or hip were found gpu_backend was set appropriately
if (gpu_backend MATCHES "^(onemkl|auto)$")
```